diag_logs - log file consolidation

sbeaver-netgate opened the pull request:
A GET argument has been added to dig_logs.php so that the log file to be displayed can be specified.

Minor edits to the file allow all log files to be displayed correctly

The tab array has been updated in this file, and in:

diag_logs_settings.php
diag_logs_vpn.php
diag_logs_filter.php
diag_logs_filter_dynamic.php
diag_logs_filter_summary.php

The last four of these files have not yet been converted to bootstrap, but the tab arrays are now compatible with this change.

The previous PR for diag_logs_settings will be deleted.

The following files are no longer required and have been removed:

diag_logs_auth.php
diag_logs_dhcp.php
diag_logs_gateways.php
diag_logs_ipsec.php
diag_logs_ntpd.php
diag_logs_openvpn.php
diag_logs_ppp.php
diag_logs_relayd.php
diag_logs_resolver.php
diag_logs_routing.php
diag_logs_wireless.php

sbeaver-netgate committed:
diag_logs - log file consolidation

A GET argument has been added to dig_logs.php so that the log file to
displayed can be specified.

Minor edits to the file allow all log files to be displayed correctly

The tab array has been updated in this file, and in:

    diag_logs_settings.php
    diag_logs_vpn.php
    diag_logs_filter.php
    diag_logs_filter_dynamic.php
    diag_logs_filter_summary.php

The last four of these files have not yet been converted to bootstrap,
but the tab arrays are now compatible with this change.

The previous PR for diag_logs_settings will be deleted.

The following files are no longer required and have been removed:

    diag_logs_auth.php
    diag_logs_dhcp.php
    diag_logs_gateways.php
    diag_logs_ipsec.php
    diag_logs_ntpd.php
    diag_logs_openvpn.php
    diag_logs_ppp.php
    diag_logs_relayd.php
    diag_logs_resolver.php
    diag_logs_routing.php
    diag_logs_wireless.php

SjonHortensius commented on usr/local/www/diag_logs.php:
```diff
@@ -67,26 +73,28 @@
-$tab_array[] = array(gettext("Resolver"), false, "/diag_logs_resolver.php");
-$tab_array[] = array(gettext("Wireless"), false, "/diag_logs_wireless.php");
-display_top_tabs($tab_array, false, 'nav nav-tabs');
+if(($logfile == 'system') || ($logfile == 'gateways') || ($logfile == 'routing') || ($logfile == 'resolver') || ($logfile == 'wireless')) {
```
You can rewrite this to something more readable, use:

``` php
if (in_array($logfile, array('system', 'gateways', 'routing' ...)))
```

sbeaver-netgate replied:
Thanks. I’ll do that.

> On May 4, 2015, at 10:12 AM, SjonHortensius notifications@github.com wrote:
> 
> In usr/local/www/diag_logs.php https://github.com/SjonHortensius/pfsense/pull/160#discussion_r29587103:
> 
> >  $tab_array[] = array(gettext("Settings"), false, "diag_logs_settings.php");
> >  display_top_tabs($tab_array);
> > 
> >  $tab_array = array();
> > -$tab_array[] = array(gettext("General"), true, "/diag_logs.php");
> > -$tab_array[] = array(gettext("Gateways"), false, "/diag_logs_gateways.php");
> > -$tab_array[] = array(gettext("Routing"), false, "/diag_logs_routing.php");
> > -$tab_array[] = array(gettext("Resolver"), false, "/diag_logs_resolver.php");
> > -$tab_array[] = array(gettext("Wireless"), false, "/diag_logs_wireless.php");
> > -display_top_tabs($tab_array, false, 'nav nav-tabs');
> > +if(($logfile == 'system') || ($logfile == 'gateways') || ($logfile == 'routing') || ($logfile == 'resolver') || ($logfile == 'wireless')) {
> > You can rewrite this to something more readable, use:
> 
> if (in_array($logfile, array('system', 'gateways', 'routing' ...)))
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/160/files#r29587103.

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com mailto:sbeaver@netgate.com | www.netgate.com http://www.netgate.com/

sbeaver-netgate committed:
diag_logs.php Minor edit per SH suggestion

Substitute in_array() for long ‘if’ statement

Thanks for the suggestion

SjonHortensius commented:
This change looks good; the only issue I see is you included diag_logs_settings.php, to which my previous comment still applies:

> The changes you have made have overwritten my previous changes; have a look at the diff @ https://github.com/SjonHortensius/pfsense/pull/144/files.
> 
> I think you should recreate your local branch from the current HEAD, then check if these changes are still necessary :)

Could you please have a look at https://github.com/SjonHortensius/pfsense/commit/3e607d0e1dddc0a9450ac979430861d602054029 and apply this diff to your file (and this PR); or see what you can apply manually?

sbeaver-netgate commented:
I thought I had deleted the original PR for this file and started again from the head file. I will double-check that.

Steve

> On May 4, 2015, at 10:22 AM, SjonHortensius notifications@github.com wrote:
> 
> This change looks good; the only issue I see is you included diag_logs_settings.php, to which my previous comment still applies:
> 
> The changes you have made have overwritten my previous changes; have a look at the diff @ https://github.com/SjonHortensius/pfsense/pull/144/files https://github.com/SjonHortensius/pfsense/pull/144/files.
> 
> I think you should recreate your local branch from the current HEAD, then check if these changes are still necessary :)
> 
> Could you please have a look at 3e607d0 https://github.com/SjonHortensius/pfsense/commit/3e607d0e1dddc0a9450ac979430861d602054029 and apply this diff to your file (and this PR); or see what you can apply manually?
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/160#issuecomment-98725508.

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com mailto:sbeaver@netgate.com | www.netgate.com http://www.netgate.com/

sbeaver-netgate committed:
dig_settings updated

Switched to my master branch, updated with pull from SH fork, updated
tab array and committed.

This version should correspond to the head version but with revised tab
array.

sbeaver-netgate commented:
Updated from head and applied changes to tab_array.

I believe this version is correct. I see your changes to the toggling system (which work very nicely)

> On May 4, 2015, at 10:22 AM, SjonHortensius notifications@github.com wrote:
> 
> This change looks good; the only issue I see is you included diag_logs_settings.php, to which my previous comment still applies:
> 
> The changes you have made have overwritten my previous changes; have a look at the diff @ https://github.com/SjonHortensius/pfsense/pull/144/files https://github.com/SjonHortensius/pfsense/pull/144/files.
> 
> I think you should recreate your local branch from the current HEAD, then check if these changes are still necessary :)
> 
> Could you please have a look at 3e607d0 https://github.com/SjonHortensius/pfsense/commit/3e607d0e1dddc0a9450ac979430861d602054029 and apply this diff to your file (and this PR); or see what you can apply manually?
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SjonHortensius/pfsense/pull/160#issuecomment-98725508.

## Steve

Stephen Beaver | +1 (512) 646-4100;239 | sbeaver@netgate.com mailto:sbeaver@netgate.com | www.netgate.com http://www.netgate.com/
